azure/defaults.go

```diff
@@ -383,6 +385,54 @@ func (p userAgentPolicy) Do(req *policy.Request) (*http.Response, error) {
 	return req.Next()
 }
 
+// NewDefaultCredential creates a new credential that uses the following methods, in order of precedence:
+// 1. Workload Identity
+// 2. Managed Identity
+// 3. Environment variables
+// 4. Azure CLI.
+func NewDefaultCredential(clientOptions *azcore.ClientOptions) (*azidentity.ChainedTokenCredential, error) {
```

Comment:
This function should read client ID and tenant ID from the AzureClusterIdentity or have a way to take this input preferably via a structure. 

I can think something like below : 

```
type CredentialOptions struct {
*azcore.ClientOptions
clientID string
tenantID string
}
```
func NewDefaultCredential(credOptions *CredentialOptions) (*azidentity.ChainedTokenCredential, error) {

}
```
Also at line no 400 we can set the tenantID and clientID in `azidentity.WorkloadIdentityCredentialOptions` options to whatever is set in `credOptions` which will fix the problem.